dipy/viz/interactor.py

```diff
@@ -0,0 +1,289 @@
+import numpy as np
+
+# Conditional import machinery for vtk
+from dipy.utils.optpkg import optional_package
+
+# Allow import, but disable doctests if we don't have vtk
+vtk, have_vtk, setup_module = optional_package('vtk')
+
+if have_vtk:
+    vtkInteractorStyleUser = vtk.vtkInteractorStyleUser
+    # version = vtk.vtkVersion.GetVTKSourceVersion().split(' ')[-1]
+    # major_version = vtk.vtkVersion.GetVTKMajorVersion()
+else:
+    vtkInteractorStyleUser = object
+
+
+class Event(object):
+    def __init__(self):
+        self.position = None
+        self.name = None
+        self.key = None
+        self._abort_flag = None
+
+    @property
+    def abort_flag(self):
+        return self._abort_flag
+
+    def update(self, event_name, interactor):
+        """ Updates current event information. """
+        self.name = event_name
+        self.position = np.asarray(interactor.GetEventPosition())
+        self.key = interactor.GetKeySym()
+        self._abort_flag = False  # Reset abort flag
+
+    def abort(self):
+        """ Aborts the event i.e. do not propagate it any further. """
+        self._abort_flag = True
+
+
+class CustomInteractorStyle(vtkInteractorStyleUser):
+    """ Manipulate the camera and interact with objects in the scene.
+
+    This interactor style allows the user to interactively manipulate (pan,
+    rotate and zoom) the camera. It also allows the user to interact (click,
+    scroll, etc.) with objects in the scene.
+
+    Several events handling methods from :class:`vtkInteractorStyleUser` have
+    been overloaded to allow the propagation of the events to the objects the
+    user is interacting with.
+
+    In summary, while interacting with the scene, the mouse events are as
+    follows:
+    - Left mouse button: rotates the camera
+    - Right mouse button: dollys the camera
+    - Mouse wheel: dollys the camera
+    - Middle mouse button: pans the camera
+    """
+    def __init__(self):
+        # Default interactor is responsible for moving the camera.
+        self.default_interactor = vtk.vtkInteractorStyleTrackballCamera()
+        # The picker allows us to know which object/actor is under the mouse.
+        self.picker = vtk.vtkPropPicker()
```

Comment:
The interactor shouldn't be the one that picks individual voxels. The interactor is responsible for picking actor2D/3D. The way I see it, picking voxels should be handled in a callback function associated to the event LeftButtonPress. So, when the user clicks somewhere on a 3D slicer, the interactor sees that as picking the "3D volume" actor then calls the associated callback function in which we have a vtkcellpicker that uses the event position to retrieve the voxel that was picked..

Also, if I remember vtkcellpicker wasn't supported on some computers maybe because of wrong vtk version or not having a dedicated graphic.